src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -71,6 +72,7 @@ export const RichTextEditor: FC<RichTextEditorProps> = ({
     const [{ matches, children }, send] = useMachine(() =>
         editorMachine.withContext({ locked: readonly, onTextChange, onBlur }),
     );
+    useEditorValueUpdates(editor, initialValue);
```

Comment:
Is it updating the xstate's context? Seems like the `editor` object is not touched here

Reply:
`editor` and `value` are not part of the xstate context (they are sent with the events)